src/V3EmitCSyms.cpp

```diff
@@ -687,6 +687,7 @@ void EmitCSyms::emitSymImp() {
 
     if (v3Global.opt.vpi()) {
         puts("\n// Setup scope hierarchy\n");
+        puts("__Vhier.clear();\n");
```

Comment:
Clearing the local __Vhier is ok, but if this clears the global table which I think it will do, this will break cases where there's multiple verilator modules in a large hierarchy sharing the same DPI table.  Unfortunately I don't think we have a test for that; t_trace_two does it for tracing but we probably should have a t_dpi_two that's similar.

Reply:
Perhaps this answer makes it clear why it is "preserved".

BTW the proper fix is probably to add simulation context structure per #2660.

Reply:
Yeah, that makes sense.  How about adding a `__Syms` destructor that removes scopes from `m_hierMap`?  Seems like that would be safe wrt multiple verilated modules.

Reply:
That seems reasonable, of course it needs to be ok to be called from multiple module's destructors which clear() would be.